test/unit/rules/lint-no-arguments-for-html-elements-test.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+const ERROR_MESSAGE = require('../../../lib/rules/lint-no-arguments-for-html-elements')
+  .ERROR_MESSAGE;
+const ERROR_MESSAGE_2 = require('../../../lib/rules/lint-no-arguments-for-html-elements')
+  .ERROR_MESSAGE_2;
+const ERROR_MESSAGE_3 = require('../../../lib/rules/lint-no-arguments-for-html-elements')
+  .ERROR_MESSAGE_3;
+
+generateRuleTests({
+  name: 'no-arguments-for-html-elements',
+
+  config: true,
+
+  good: [
+    '<Input @name=1 />',
+    '{{#let (component \'foo\') as |bar|}} <bar @name="1" as |n|><n/></bar> {{/let}}',
+    '<@externalComponent />',
+    `<MyComponent>
+    <:slot @name="Header"></:slot>
+  </MyComponent>`,
+    '<@foo.bar @name="2" />',
+    '<@foo @name="2" />',
+    '<foo.some.name @name="1"/>',
+  ],
+
+  bad: [
+    {
+      template: '<div as |a|></div>',
```

Comment:
this looks unrelated to a rule called `no-arguments-for-html-elements`. IMHO we should extract that into a `no-block-params-for-html-elements` rule.

Reply:
gotta say I'm surprised that this got merged without splitting it into several rules, especially now that it's planned to be included in the recommended ruleset...

Reply:
hmmm, ya I must have missed this. Good catch, we should absolutely split it out.

Reply:
Made https://github.com/ember-template-lint/ember-template-lint/issues/1451 to track, and added to v3.0.0 milestone.